.github/workflows/build-test.yml

```diff
@@ -13,13 +18,21 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          fetch-depth: 0
+          fetch-depth: 1
 
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
           go-version: 1.18
 
+      - name: Cache go mods
```

Comment:
Instead of this can we upgrade setup-go https://github.com/actions/setup-go?tab=readme-ov-file#v4 it says it will cache in default what do you think?

Reply:
Yes, you are right upgrading is the better option. Updated.